arrow/parser.py

```diff
@@ -175,7 +175,7 @@ def _parse_token(self, token, value, parts):
         elif token in ['DD', 'D']:
             parts['day'] = int(value)
 
-        elif token in ['HH', 'H']:
+        elif token in ['HH', 'H', 'hh', 'h']:
```

Comment:
Updated that as per your comment, however I can't use the same logic for dd/d since they can't be parsed the same way as DD/D (ints VS names). I'll give that a shot in a separate pull request.